### DISCUSSION TITLE
Learn the PowerShell engine

### DISCUSSION THREAD

**Discussion by FranciscoNabas** [PowerShell/PowerShell#19424] [Q&A] [answered]
Hi everyone!
My name is Francisco, and I'm a PowerShell fan.
I'm asking this question here, because it's surprisingly a difficult topic to find information about. 
I've been studying, building modules, writing blog posts for a while, and I really want to learn how the main PowerShell engine works, and start experimenting.
Is there any good resource, where I can learn how it works, and how to interact with it?
I'm not a professional developer, but I have general knowledge in C#, C++, and, well... PowerShell.
I've cloned the repo, and I'm literally reading the code, which helps giving an idea, but I don't expect being able to read it all and fully understand it by myself.
I hope I made sense, and sorry if this isn't the place for this question.
Hope you all have a great week!

**Comment by iSazonov** [comment, marked as answer]
https://www.youtube.com/watch?v=3fL4GG0dvkE

**Reply by FranciscoNabas** [reply]
That video it's awesome. That gave a good overall to start with. Thank you very much!

**Comment by Ayanmullick** [comment]
@iSazonov, I couldn't repair the file below. Is there another location?
https://github.com/psconfeu/2018/tree/master/Dongbo%20Wang  

I wish there was a [code walkthrough](https://www.youtube.com/watch?v=8PVufdYi2zs) too. I could only find a repository visualizer.

https://mango-dune-07a8b7110.1.azurestaticapps.net/?repo=PowerShell%2FPowerShell

**Reply by FranciscoNabas** [reply]
Was able to get it from here.

https://github.com/psconfeu/2018/blob/master/Dongbo%20Wang/psconfeu_2018_EngineInternals.pptx

**Reply by Ayanmullick** [reply]
Yes, that worked.
